pkg/execution/executor/executor.go

```diff
@@ -2346,6 +2372,31 @@ func (e *executor) handleGeneratorWaitForEvent(ctx context.Context, gen state.Ge
 	}
 
 	opcode := gen.Op.String()
+	now := time.Now()
+
+	ctx, span := telemetry.NewSpan(ctx,
+		telemetry.WithScope(consts.OtelScopeWait),
+		telemetry.WithName(consts.OtelSpanWaitForEvent),
+		telemetry.WithTimestamp(now),
+		telemetry.WithSpanAttributes(
+			attribute.Bool(consts.OtelUserTraceFilterKey, true),
```

Comment:
This will need some attributes adding to fulfil the [resolvers](https://github.com/inngest/monorepo/blob/552fbeceaadf2f065d80215d59b58fc9984097df/api/schema.graphql#L1028-L1034), namely:
- Event name
- Expression
- Expiry time